src/mjolnir/adminbuilder.cc

```diff
@@ -645,45 +645,44 @@ bool BuildAdminFromPBF(const boost::property_tree::ptree& pt,
   }
   LOG_INFO("Created Level index");
 
-  sql = "CREATE INDEX IdxDriveOnRight ON admins (drive_on_right)";
+  sql = "UPDATE admins AS child SET parent_admin = (SELECT parent.rowid from admins";
+  sql +=
+      " AS parent WHERE parent.admin_level < child.admin_level AND ST_Covers(parent.geom, child.geom) ";
+  sql += " ORDER BY parent.admin_level ASC LIMIT 1)";
```

Comment:
The diff is horribly incomprehensible, but it's simple: do the `parent_admin` update first, then `drive_on_right`, then `allow_intersection_names`. 

In this query the where clause ordering is key: first filter by level (we only care about rows with admin_level < current level), for which we already have an index, then we only have to run `ST_Covers` for admins with smaller admin levels.